harness/propertyHelper.js

```diff
@@ -1,13 +1,17 @@
 
 function isConfigurable(obj, name) {
+    var descriptor = Object.getOwnPropertyDescriptor(obj, name);
     try {
         delete obj[name];
     } catch (e) {
         if (!(e instanceof TypeError)) {
             $ERROR("Expected TypeError, got " + e);
         }
     }
-    return !Object.prototype.hasOwnProperty.call(obj, name);
+    var result = !Object.prototype.hasOwnProperty.call(obj, name);
+    // Restore property (no-op if not deleted)
+    Object.defineProperty(obj, name, descriptor);
```

Comment:
I changed this to have a `restore` option instead, which is doing exactly what you're doing here, but now in the new API: https://github.com/tc39/test262/pull/979